workbench-core/accounts/src/services/projectService.ts

```diff
@@ -127,91 +141,58 @@ export default class ProjectService {
-    // welcome to party town
+    // Else, member of more than 1 group, batch get items and manually filter, sort, and paginate
     const projectIds: string[] = userGroupsForCurrentUser.map(
       (projectGroup: string) => projectGroup.split('#')[0]
     );
-
-    // i = 0
-    let filterExpression = 'id = :proj1';
-    const expressionAttributeValues: { [key: string]: string } = { ':proj1': projectIds[0] };
-    for (let i = 1; i < projectIds.length; i++) {
-      // build filter string "id = :proj1 OR id = :proj2 OR ... OR id = projN+1"
-      filterExpression = filterExpression + ` OR id = :proj${i + 1}`;
-      // build expressionAttributeValues object {":proj1": "projectId1", ..., ":projN+1": "projectIdN"}
-      expressionAttributeValues[`:proj${i + 1}`] = projectIds[i];
-    }
-
-    // TODO--extend to other GSIs after filtering is added
-    let queryParams: QueryParams = {
-      key: { name: 'resourceType', value: this._resourceType },
-      index: 'getResourceByCreatedAt',
-      limit: pageSize,
-      filter: filterExpression,
-      values: expressionAttributeValues
-    };
-
-    let projects: Project[] = [];
-
-    let returnedCount = 0;
-    while (returnedCount < pageSize && returnedCount < projectIds.length) {
-      queryParams = addPaginationToken(paginationToken, queryParams);
-
-      const projectsResponse = await this._aws.helpers.ddb.query(queryParams).execute();
-
-      returnedCount += projectsResponse.Count!;
-
-      // nothing returned
-      if (projectsResponse.Items === undefined || _.isEmpty(projectsResponse.Items)) {
-        paginationToken = getPaginationToken(projectsResponse);
-        console.log(paginationToken);
-      }
-
-      // nothing else to get from DDB
-      else if (projectsResponse.LastEvaluatedKey === undefined) {
-        projects = projects.concat(projectsResponse.Items.map((item) => this._formatFromDDB(item)));
-        paginationToken = undefined;
+    const keys: { [key: string]: unknown }[] = projectIds.map((projectId) =>
+      buildDynamoDBPkSk(projectId, resourceTypeToKey.project)
+    );
+    const projectsResponse = (await this._aws.helpers.ddb.get(keys).execute()) as BatchGetItemCommandOutput;
+    if (projectsResponse.Responses) {
+      const projects: Project[] = projectsResponse.Responses[this._tableName].map((item) =>
+        this._formatFromDDB(item)
+      );
+      // apply sort or filter
+      let projectsOnPage: Project[] = projects;
+      if (filter) {
+        projectsOnPage = manualFilterProjects(filter, projects);
       }
-
-      // too little--while loop continues
-      else if (returnedCount < pageSize) {
-        projects = projects.concat(projectsResponse.Items.map((item) => this._formatFromDDB(item)));
-        paginationToken = getPaginationToken(projectsResponse);
+      if (sort) {
+        projectsOnPage = manualSortProjects(sort, projects);
+      } else if (filter === undefined) {
```

Comment:
should this be `if (sort === undefined)`?

Reply:
if so, can be replaced with just an `else`

Reply:
this third condition is if filter and sort are both not defined. We need to handle the cases:
1. filter is defined --> apply filter
2. sort is defined --> apply sort
3. filter and sort are defined --> apply filter and then sort (there are checks above that verify the attribute to filter and sort on are the same) so no `else if` between the first two checks
4. filter and sort are undefined --> apply default sort.

So it cannot just be an `else` because filter might have been defined above. The `else` part of the `else if` will be true when sort is undefined and the `if` condition of this checks to make sure filter was also undefined.

Reply:
I think we should remove the `else if` and breaking that login into its own conditional would be best for readability

Reply:
Yeah, that works as well and probably does increase readability. Will add!